clients/src/main/java/org/apache/kafka/clients/consumer/KafkaConsumer.java

```diff
@@ -1167,11 +1172,17 @@ public void wakeup() {
         this.client.wakeup();
     }
 
-    private void close(boolean swallowException) {
+    private void close(long timeoutMs, boolean swallowException) {
         log.trace("Closing the Kafka consumer.");
-        AtomicReference<Throwable> firstException = new AtomicReference<Throwable>();
+        AtomicReference<Throwable> firstException = new AtomicReference<>();
         this.closed = true;
-        ClientUtils.closeQuietly(coordinator, "coordinator", firstException);
+
+        try {
+            coordinator.close(timeoutMs);
+        } catch (Exception e) {
+            log.error("Failed to close coordinator", e);
```

Comment:
Alternatively, should we add a `ClientUtils.closeQuietly()` overload that accepts a timeout as well? If we're eventually going to make this method correctly respect the timeout, we're going to need to have all these close methods accept timeout. Maybe some are guaranteed to finish quickly, but some have `await` calls in them, e.g. the `Metrics` close method. Might want to introduce something like a `TimeoutCloseable` interface in addition to `Closeable` to support this generally.